### PR TITLE
external/webserver: Fix svace WGID 230304 webserver memory leak

### PR DESCRIPTION
**Pull request by amitpurwar**
Signed-off-by: Amit Purwar <amit.purwar@samsung.com>

### CODE REVIEW

File: external/webserver/http_client.c [hunk 1]
```diff
@@ -1150,6 +1150,7 @@ int http_send_response_helper(struct http_client_t *client, int status, const ch
                                                            body_len);
 				if (len < 0) {
 					HTTP_LOGE("Error: snprintf failed \n");
+					HTTP_FREE(buf);
```
Comment (abhishek-samsung) [inline external/webserver/http_client.c:1153]: how about adding goto for the error cases?
```
goto error;
```
```
error :
	HTTP_FREE(buf);
	return HTTP_ERROR;
```

Comment (amitpurwar) [reply external/webserver/http_client.c:1153]: I prefer to avoid using "goto" statement.

**Review by j11591 — APPROVED** [review]
LGTM

**Review by vibhor-m — COMMENTED** [review]
changes are fine

**Pull request merged**